src/Oro/Bundle/WorkflowBundle/Controller/Api/Rest/WorkflowController.php

```diff
@@ -173,6 +194,64 @@ public function transitAction(WorkflowItem $workflowItem, $transitionName)
     }
 
     /**
+     * Returns:
+     * - HTTP_OK (200) response: array('workflowItem' => array('id' => int, 'result' => array(...), ...))
+     * - HTTP_BAD_REQUEST (400) response: array('message' => errorMessageString)
+     * - HTTP_FORBIDDEN (403) response: array('message' => errorMessageString)
+     * - HTTP_NOT_FOUND (404) response: array('message' => errorMessageString)
+     * - HTTP_INTERNAL_SERVER_ERROR (500) response: array('message' => errorMessageString)
+     *
+     * @Rest\Get(
+     *      "/api/rest/{version}/workflow/transit/{entityName}/{entityId}/{transitionName}",
+     *      requirements={"version"="latest|v1", "entityId"="\d+"},
+     *      defaults={"version"="latest", "_format"="json"}
+     * )
+     * @ApiDoc(description="Perform transition for workflow item by entity class and id", resource=true)
+     * @AclAncestor("oro_workflow")
+     *
+     * @param string $entityName The class name, url-safe class name, alias or plural alias of the entity
+     * @param int    $entityId
+     * @param string $transitionName
+     *
+     * @return Response
+     */
+    public function transitByEntityAction($entityName, $entityId, $transitionName)
+    {
+        try {
+            /** @var EntityRoutingHelper $routingHelper */
+            $routingHelper = $this->get('oro_entity.routing_helper');
+
+            try {
+                $entity = $routingHelper->getEntity($entityName, $entityId);
+            } catch (\ReflectionException $e) {
+                throw new BadRequestHttpException($e->getMessage(), $e);
+            }
+
+            /** @var WorkflowManager $workflowManager */
+            $workflowManager = $this->get('oro_workflow.manager');
+
+            $workflowItem = $workflowManager->getWorkflowItemByEntity($entity);
+            if (!$workflowItem) {
+                throw new NotManageableEntityException($routingHelper->resolveEntityClass($entityName));
```

Comment:
Please, cover this case with test. I don't think here should be this kind of exception, I think BadRequestHttpException will be better.